clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -107,6 +339,15 @@ public ConfigDef define(String name, Type type, Importance importance, String do
         return define(name, type, NO_DEFAULT_VALUE, null, importance, documentation);
     }
 
+
+    /**
+     * Mark the end of the configuration definition.
+     */
+    public ConfigDef end() {
```

Comment:
An alternative to requiring an explicit call to `end()` would be to call `ensureDependentsDefined` at the beginning of `parse`/`validate`.